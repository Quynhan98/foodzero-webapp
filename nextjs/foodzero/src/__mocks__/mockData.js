export const AVATAR_MOCK =
  'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fjohn-doe.webp?alt=media&token=6963f744-1ab8-4b09-971d-2486cd5375bf'

export const PRICE_LIST_MOCK = {
  name: 'Deep Sea Snow White Cod Fillet',
  price: 20,
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. ',
}

export const CARD_BLOG = {
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2FFeature%20Image.webp?alt=media&token=766a34ea-32e0-40e7-90e6-299f9af22eb1',
  title: 'Fruit and vegetables and protection against diseases',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
  category: 'Fashion',
  srcAvatar: AVATAR_MOCK,
  name: 'Julie Christie',
  date: '2022-12-06T23:29:35.000Z',
  numberOfComments: 2,
}

export const CARD_FEATURE = {
  title: 'Premium Quality',
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2FCircleIcon.webp?alt=media&token=1df373f3-27a3-45f6-82fa-d997c47c718f',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Neque congue arcu',
}

export const CARD_CATEGORY = {
  href: '/',
  category: 'Starters',
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fcategory.webp?alt=media&token=9f307d7c-b242-479d-943a-4ae672ac601e',
}

export const USERS_MOCK = [
  {
    id: 'gff1',
    email: 'foodzero@gmail.com',
    password: '123456789',
  },
]

export const QUOTE_MOCK = [
  {
    id: '1',
    authorName: 'John Doe',
    job: 'Blogger',
    quote:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus impedient. Turpis egestas ultricies purus  Lorem ipsum dolor sit amet.',
    avatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fjohn-doe.webp?alt=media&token=6963f744-1ab8-4b09-971d-2486cd5375bf',
  },
  {
    id: '2',
    authorName: 'Julie Christie',
    job: 'Blogger',
    quote:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus impedient. Turpis egestas ultricies purus  Lorem ipsum dolor sit amet1.',
    avatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Funnamed.webp?alt=media&token=72862fa3-6567-4043-aab0-cc449b30e602',
  },
  {
    id: '3',
    authorName: 'Dianne Russell',
    job: 'Blogger',
    quote:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus impedient. Turpis egestas ultricies purus  Lorem ipsum dolor sit amet2.',
    avatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fcute-avatar.webp?alt=media&token=7b6d918e-3dec-43cf-a4bd-70afdbdc63e0',
  },
]

export const OUR_MENU_MOCK = [
  PRICE_LIST_MOCK,
  PRICE_LIST_MOCK,
  PRICE_LIST_MOCK,
  PRICE_LIST_MOCK,
]

export const FEATURES_MOCK = [CARD_FEATURE, CARD_FEATURE, CARD_FEATURE]

export const BLOG_SECTION_MOCK = [CARD_BLOG, CARD_BLOG]

export const CATEGORY_SECTION_MOCK = [
  CARD_CATEGORY,
  CARD_CATEGORY,
  CARD_CATEGORY,
]
