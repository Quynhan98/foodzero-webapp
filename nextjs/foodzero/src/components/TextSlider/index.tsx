import { memo, useState } from 'react'
import Image from 'next/image'
import {
  Box,
  BoxProps,
  Button,
  Flex,
  Text,
  useMediaQuery,
} from '@chakra-ui/react'

// Components
import UserAvatar from '@components/UserAvatar'

// Constants
import { BREAKPOINTS } from '@constants/index'

type QuoteType = {
  id: string
  authorName: string
  job: string
  quote: string
  avatar: string
}

interface TextSliderProps extends BoxProps {
  quotes: QuoteType[]
}

const TextSlider = ({ quotes, ...rest }: TextSliderProps) => {
  const [isMobile] = useMediaQuery(BREAKPOINTS.MEDIUM)
  const [quote, setQuote] = useState(quotes[0])
  const [currentSlider, setCurrentSlider] = useState<number>(1)

  const handleGoBack = () => {
    const newCurrentSlider = currentSlider > 1 ? currentSlider - 1 : 1
    setCurrentSlider(newCurrentSlider)

    const currentQuote = quotes[newCurrentSlider - 1]
    setQuote(currentQuote)
  }

  const handleNext = () => {
    const newCurrentSlider =
      currentSlider < quotes.length ? currentSlider + 1 : currentSlider
    setCurrentSlider(newCurrentSlider)

    const currentQuote = quotes[newCurrentSlider - 1]
    setQuote(currentQuote)
  }

  return (
    <Box {...rest}>
      <Text
        size={{ base: 'default', md: 'extraLarge' }}
        variant="secondary"
        fontFamily="Rufina"
        maxW="1262px"
      >
        â€œ {quote.quote}
      </Text>
      <Flex
        flexDirection={{ base: 'column', md: 'row' }}
        alignItems="center"
        justifyContent="space-between"
        pt={{ base: '20px', md: '86px' }}
        gap={{ base: '15px', md: '10px' }}
      >
        <UserAvatar
          src={quote.avatar}
          name={quote.authorName}
          job={quote.job}
        />
        <Flex
          alignItems="center"
          width={{ base: '100%', md: '40%' }}
          justifyContent="flex-end"
        >
          <Button
            data-testid="buttonBack"
            onClick={handleGoBack}
            variant="light"
          >
            <Image
              width={isMobile ? 23 : 46}
              height={isMobile ? 13 : 26}
              src="/images/leftArrow.webp"
              alt="right arrow icon"
            />
          </Button>
          <Text
            fontFamily="rufina"
            fontSize={{ base: 'base', md: 'md' }}
            lineHeight={{ base: 'base', md: 'sm' }}
            variant="secondary"
          >
            {currentSlider}/{quotes.length}
          </Text>
          <Button data-testid="buttonNext" onClick={handleNext} variant="light">
            <Image
              width={isMobile ? 23 : 46}
              height={isMobile ? 13 : 26}
              src="/images/rightArrow.webp"
              alt="right arrow icon"
            />
          </Button>
        </Flex>
      </Flex>
    </Box>
  )
}

export default memo(TextSlider)
