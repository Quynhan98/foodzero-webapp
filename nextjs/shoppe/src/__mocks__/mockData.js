export const PRODUCT_ITEM = {
  id: 1,
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/test-1-50371.appspot.com/o/shoppe-products%2FImg-04.png?alt=media&token=641c3cfe-cab6-4498-8632-eb511434e042',
  name: 'Lira Earrings',
  price: 20,
  status: 'Sold out',
}
