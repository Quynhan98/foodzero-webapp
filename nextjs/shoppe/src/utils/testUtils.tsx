import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>{component}</ChakraProvider>,
  )
}

export * from '@testing-library/react'
export { customRender as render }
