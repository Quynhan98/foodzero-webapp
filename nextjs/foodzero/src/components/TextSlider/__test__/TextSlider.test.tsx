import { fireEvent, render, screen } from '@utils/testUtils'

// Components
import TextSlider from '@components/TextSlider'

// Mocks
import { QUOTE_MOCK } from '@mocks/mockData'

describe('TextSlider component', () => {
  const props = {
    quotes: QUOTE_MOCK,
  }

  it('Renders TextSlider correctly', () => {
    const { container } = render(<TextSlider {...props} />)

    expect(container).toMatchSnapshot()
  })

  it('Renders current slide and click button correctly', () => {
    render(<TextSlider {...props} />)

    const buttonNext = screen.getByTestId('buttonNext')
    const buttonBack = screen.getByTestId('buttonBack')

    expect(screen.getByText(`1/${QUOTE_MOCK.length}`)).toHaveTextContent(
      `1/${QUOTE_MOCK.length}`,
    )

    fireEvent.click(buttonNext)
    expect(screen.getByText(`2/${QUOTE_MOCK.length}`)).toHaveTextContent(
      `2/${QUOTE_MOCK.length}`,
    )

    fireEvent.click(buttonBack)
    expect(screen.getByText(`1/${QUOTE_MOCK.length}`)).toHaveTextContent(
      `1/${QUOTE_MOCK.length}`,
    )
  })
})
