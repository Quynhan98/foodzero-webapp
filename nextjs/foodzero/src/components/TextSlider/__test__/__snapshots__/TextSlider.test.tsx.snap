// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSlider component Renders TextSlider correctly 1`] = `
<div>
  <div
    class="css-0"
  >
    <p
      class="chakra-text css-zelfmh"
    >
      â€œ 
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus impedient. Turpis egestas ultricies purus  Lorem ipsum dolor sit amet.
    </p>
    <div
      class="css-1vjcr4l"
    >
      <div
        class="css-mgvg8t"
      >
        <span
          class="chakra-avatar css-1kva3cr"
        >
          <div
            aria-label="John Doe"
            class="chakra-avatar__initials css-uo84k3"
            role="img"
          >
            JD
          </div>
        </span>
        <div
          class="css-0"
        >
          <h5
            class="chakra-heading css-ai6zdm"
          >
            John Doe
          </h5>
          <p
            class="chakra-text css-lv0zmh"
          >
            Blogger
          </p>
        </div>
      </div>
      <div
        class="css-11yd5ay"
      >
        <button
          class="chakra-button css-yvsqkp"
          data-testid="buttonBack"
          type="button"
        >
          <img
            alt="right arrow icon"
            data-nimg="1"
            decoding="async"
            height="26"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2FleftArrow.webp&w=96&q=75"
            srcset="/_next/image?url=%2Fimages%2FleftArrow.webp&w=48&q=75 1x, /_next/image?url=%2Fimages%2FleftArrow.webp&w=96&q=75 2x"
            style="color: transparent;"
            width="46"
          />
        </button>
        <p
          class="chakra-text css-37ekto"
        >
          1
          /
          3
        </p>
        <button
          class="chakra-button css-yvsqkp"
          data-testid="buttonNext"
          type="button"
        >
          <img
            alt="right arrow icon"
            data-nimg="1"
            decoding="async"
            height="26"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2FrightArrow.webp&w=96&q=75"
            srcset="/_next/image?url=%2Fimages%2FrightArrow.webp&w=48&q=75 1x, /_next/image?url=%2Fimages%2FrightArrow.webp&w=96&q=75 2x"
            style="color: transparent;"
            width="46"
          />
        </button>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
